src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/mappers/AmendmentToEdifactMapper.java

```diff
@@ -0,0 +1,49 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment.mappers;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.PatchValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AmendmentToEdifactMapper {
+
+    protected static boolean amendmentPatchRequiringValue(AmendmentPatch amendmentPatch) {
+        return amendmentPatch.getOp() == AmendmentPatchOperation.ADD
+            || amendmentPatch.getOp() == AmendmentPatchOperation.REPLACE;
+    }
+
+    public Optional<Segment> map(AmendmentBody amendmentBody) throws PatchValidationException {
+        var patches = amendmentBody.getJsonPatches();
+        validatePatches(patches);
+        if (shouldCreateSegment(amendmentBody)) {
+            return Optional.of(mapPatches(amendmentBody));
+        }
+        return Optional.empty();
```

Comment:
Using Optional with functional approach would make `shouldCreateSegment()` unnecessary. Almost all segments will be optional so it's just masked `if (segment.isPresent()) {}`

Reply:
It's more complex in other mappers. In this way it separates decision from mapping logic. Let's leave as is for now and we'll see how it goes